components/content_settings/core/browser/brave_content_settings_pref_provider.cc

```diff
@@ -341,6 +343,50 @@ void BravePrefProvider::MigrateShieldsSettingsV1ToV2() {
   prefs_->SetInteger(kBraveShieldsSettingsVersion, 2);
 }
 
+void BravePrefProvider::MigrateDefaultHttpsContentSetting() {
+  auto rule_iterator = PrefProvider::GetRuleIterator(
+      ContentSettingsType::BRAVE_HTTP_UPGRADABLE_RESOURCES,
+      /*off_the_record*/ false);
+
+  using OldRule = std::pair<ContentSettingsPattern, ContentSettingsPattern>;
+  std::vector<OldRule> old_rules;
+  const ContentSettingsPattern& wildcard = ContentSettingsPattern::Wildcard();
+  std::vector<Rule> new_rules;
+  base::Value default_value;
+  // Find rules that can be migrated and create replacement rules for them.
+  while (rule_iterator && rule_iterator->HasNext()) {
+    auto old_rule = rule_iterator->Next();
+    if (old_rule.primary_pattern == wildcard &&
+        old_rule.secondary_pattern == wildcard) {
+      default_value = old_rule.value.Clone();
+      continue;
+    }
+
+    Rule new_rule;
+    new_rule.expiration = old_rule.expiration;
+    new_rule.session_model = old_rule.session_model;
+    new_rule.value = std::move(old_rule.value);
+    new_rule.secondary_pattern = old_rule.secondary_pattern;
+    new_rule.primary_pattern = old_rule.primary_pattern;
+    new_rules.push_back(std::move(new_rule));
+  }
+  if (default_value.is_none())
+    return;
+  rule_iterator.reset();
+  ClearAllContentSettingsRules(
```

Comment:
why are we clearing all of them and then resetting? We only need to delete the default rule which we can do by setting it to CONTENT_SETTING_DEFAULT

Reply:
Tried to set CONTENT_SETTING_DEFAULT, it doesnt remove wildcard rule `"*,*"` from the exceptions list

Reply:
that's very strange and makes me wonder if we're doing something wrong here